pywinauto/timings.py

```diff
@@ -319,11 +319,45 @@ def WaitUntil(
             
     return func_val
 
+#=========================================================================
+def wait_until(
+    timeout, 
+    retry_interval, 
+    value = True, 
+    op = operator.eq):
+    
+    """
+	WaitUntil decorator 
+    """
+    
+    def wait_until_decorator(func):		
+        def wrapped(*args):
+	
+            start = time.time()
 
+            func_val = func(*args)
+            # while the function hasn't returned what we are waiting for    
+            while not op(func_val, value):         
+                # find out how much of the time is left
+                time_left = timeout - ( time.time() - start)
+    
+                # if we have to wait some more        
+                if time_left > 0:
+                    # wait either the retry_interval or else the amount of
+                    # time until the timeout expires (whichever is less)
+                    time.sleep(min(retry_interval, time_left))
+                    func_val = func(*args)
+                else:
+                    err = TimeoutError("timed out")
+                    err.function_value = func_val
+                    raise err            
+            return func_val			
+        return wrapped
+    return wait_until_decorator
+	

```

Comment:
I don't like the copy-paste. The code has been duplicated from the original ```WaitUntil``` function. The core implementation should be kept at only one place.
```python
def WaitUntill(timeout, 
               retry_interval, 
               func, 
               value = True, 
               op = operator.eq,
               *args):

    return wait_until(timeout, retry_interval, value, op)(func)(*args)
    

timeout = 1.0
retry_interval = 0.3

@wait_until(timeout, retry_interval)
def decorated_foo():
    print("decorated_foo")
    return True  # False

def foo():
    print("foo")
    return True  # False

# Call as usual
WaitUntill(timeout, retry_interval, foo)

# Call a decorated func
decorated_foo()
```